### DISCUSSION TITLE
External LoadBalancing/IP overwrite for gateway-service

### DISCUSSION THREAD

**Discussion by stebenz** [linkerd/linkerd2#4910] [General]
Is there a possibility to provide the Linkerd gateway-service with the external IP without the requirement to have a kubernetes-service type LoadBalancer, where the IP is read from ".status.loadBalancer.ingress" and so only available if the service is type LoadBalancer?

The idea was to build a multi-regional cockroachdb with the help of Linkerd and multiple clusters based on Google Cloud VMs. The clusters each have a running ambassador with a service type NodePort where an external load balancer handles the traffic to the nodes, which share 1 external IP as a cluster (to be exact 2 as the kubeapi has a separate IP).

So Linkerd doesn’t need to support any external load balancer, but it would have to support another way to provide the external IP besides the requirement to support service with type LoadBalancer.

I also found some issues regarding this topic but these use external DNS entries, if I understand correctly, which we want to avoid:
- https://github.com/linkerd/linkerd2/issues/4584
- https://github.com/linkerd/linkerd2/issues/4582

Thanks!

**Comment by ihcsim** [comment]
@stebenz The recent [edge releases](https://github.com/linkerd/linkerd2/releases/tag/edge-20.8.4) have some new additions to multicluster which may match what you are looking for. Specifically, there is a new `Link` CRD with a `spec.gatewayAddress` field which you can use to point to your existing in-cluster Ambassador. Some additional configuration will be needed on your Ambassadors. (E.g., it needs to be injected with the Linkerd proxy.) See the multicluster doc on [leveraging Ambassador](https://linkerd.io/2/tasks/installing-multicluster/#leverage-ambassador) for more info.

Let me know if this is what you are looking for.

**Reply by stebenz** [reply]
@ihcsim Thanks for this tip, if I understand correctly this CRD should do the trick or at least should provide an option to use linkerd in our constellation. 
All the changes for leveraging Ambassador are already in place, but thanks for the hint anyway.

If I find the time, I will try out one of the edge releases to see if it would work.
